ci/clang-tidy-all.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+set -e
+
+SCRIPT=$(readlink -f "$0")
+SCRIPTPATH=$(dirname "${SCRIPT}")
+ROOTPATH=$(realpath "${SCRIPTPATH}"/..)
+if ! command -v clang-tidy; then
+    echo "clang-tidy is not found!"
+    exit 1
+fi
+
+# Determine the mode (all files or changed files)
+MODE=${1:-all}
+BUILD_DIR=${2:-build}
+
+if [ "$MODE" = "changed" ]; then
+    # Get the list of changed files from origin/dev
+    files=$(git diff --name-only origin/dev -- '*.cpp' '*.h' | grep -v '3rdParty/' || true)
+else
+    # Find all relevant files
+    files=$(find "${ROOTPATH}" -type f \( -name '*.cpp' -o -name '*.h' \) -not -path "*/3rdParty/*")
+fi
+
+# Check if there are any files to process
+if [ -z "$files" ]; then
+    echo "No files to process."
+    exit 0
+fi
+
+# Process each file
+echo "$files" | while IFS= read -r file; do
+    echo "Checking: $file"
+    clang-tidy "$file" -p $BUILD_DIR --fix --checks=modernize-use-nullptr,modernize-use-override,performance-unnecessary-value-param
```

Comment:
Do we want to install `clang-tidy` as a python package like we do for `cmake-format` and `clang-format`?
That way we may be able to control the `clang-tidy` version we're using.

Also - do we still need to install `cmake-format` as part of our CI?

Reply:
fixed